pkgs/development/python-modules/opentelemetry-instrumentation-sqlalchemy/default.nix

```diff
@@ -0,0 +1,44 @@
+{
+  buildPythonPackage,
+  pythonOlder,
+  pytestCheckHook,
+  hatchling,
+  sqlalchemy_1_4,
+  aiosqlite,
+  opentelemetry-api,
+  opentelemetry-instrumentation,
+  opentelemetry-semantic-conventions,
+  opentelemetry-test-utils,
+}:
+
+buildPythonPackage {
+  inherit (opentelemetry-instrumentation) version src;
+  pname = "opentelemetry-instrumentation-sqlalchemy";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  sourceRoot = "${opentelemetry-instrumentation.src.name}/instrumentation/opentelemetry-instrumentation-sqlalchemy";
+
+  build-system = [ hatchling ];
+
+  dependencies = [
+    sqlalchemy_1_4
```

Comment:
Please don't propagate such a versioned package.
It causes a package conflict due to `PYTHONPATH`.

> Only unversioned attributes (e.g. pydantic, but not pypdantic_1) can be included in dependencies, since due to PYTHONPATH limitations we can only ever support a single version for libraries without running into duplicate module name conflicts.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

Reply:
I don't think we're able to add this package then, the tests package fail with sqlalchemy ^2.0 currently.